### PR TITLE
Jgfouca/upstream merge 2021 09 16

### PR DESCRIPTION
**Pull request by jgfouca**
Conflict log:

### File: cime_config/allactive/config_compsets.xml

```
<<<<<<< HEAD
<!-- SCREAM DYAMOND compsets -->
<compset>
  <alias>A_SCREAM-HR-DYAMOND2</alias>
  <lname>2010_EAM%SCREAM-HR-DYAMOND2_ELM45%SPBC_MPASSI_MPASO_MOSART_SGLC_SWAV</lname>
</compset>



<!-- E3SM backup compsets -->
<compset>
 <alias>A_WCYCL1850_v0atm</alias>
 <lname>1850_EAM_ELM%SP_MPASSI_MPASO_MOSART_SGLC_SWAV</lname>
</compset>

<compset>
 <alias>A_WCYCL2000_v0atm</alias>
 <lname>2000_EAM_ELM%SP_MPASSI_MPASO_MOSART_SGLC_SWAVi</lname>
</compset>

<compset>
 <alias>A_WCYCL1850S_v0atm</alias>
 <lname>1850SOI_EAM_ELM%SP_MPASSI_MPASO_MOSART_SGLC_SWAV</lname>
</compset>

<compset>
 <alias>A_WCYCL2000S_v0atm</alias>
 <lname>2000SOI_EAM_ELM%SP_MPASSI_MPASO_MOSART_SGLC_SWAVi</lname>
</compset>


=======
>>>>>>> upstream/master
```

Resolution: Keep A_SCREAM-HR-DYAMOND2 only

### File: components/eam/cime_config/config_component.xml

```
<<<<<<< HEAD

      <value compset="AR97_EAM%SCAM" >-phys default -scam -nlev 72 -shoc_sgs -microphys p3 -rad rrtmgp -chem linoz_mam4_resus_mom_soag -rain_evap_to_coarse_aero -bc_dep_to_snow_updates</value>

=======
      <value compset="AR97_EAM%SCAM"   >-phys default -scam -nlev 72 -clubb_sgs -chem linoz_mam4_resus_mom_soag -rain_evap_to_coarse_aero -bc_dep_to_snow_updates</value>
>>>>>>> upstream/master
```

Resolution: take HEAD but cleanup whitespace

### File: components/eam/cime_config/config_compsets.xml

```
<<<<<<< HEAD
    <value  compset="_EAM%SCREAM-LR-DYAMOND1">2016-08-01</value>
    <value  compset="_EAM%SCREAM-HR-DYAMOND1">2016-08-01</value>
    <value  compset="_EAM%SCREAM-LR-DYAMOND2">2020-01-20</value>
    <value  compset="_EAM%SCREAM-HR-DYAMOND2">2020-01-20</value>
=======
        <value  compset="RICO_EAM">2004-12-16</value>
>>>>>>> upstream/master
```

Resolution: Take both

File: components/eam/src/physics/rrtmgp/external (submodule)

```
-Subproject commit 36352d774825997de083dd3b5976355bd53b0587
+Subproject commit 54ea4a9a203381868f181eb2ab88f3663160f286
```

Resolution: Take 54ea4a9a; it appears to be ahead of the commit 36352d77

### File: components/elm/bld/namelist_files/namelist_defaults.xml

```
<fsurdat hgrid="ne30np4.pg2"   sim_year="2010" use_crop=".false." >
<<<<<<< HEAD
lnd/clm2/surfdata_map/surfdata_ne30np4.pg2_simyr2010_c201210.nc</fsurdat>
=======
lnd/clm2/surfdata_map/surfdata_ne30pg2_simyr2000_c210402.nc</fsurdat>
>>>>>>> upstream/master
```

Resolution: Take upstream/master. I'm not 100% sure about this one, but I went through SCREAM's history and we don't appear to have ever touched this line. Wuyin did add the one below it to SCREAM.

### File: components/elm/src/cpl/lnd_comp_mct.F90

```
<<<<<<< HEAD
    use elm_varctl       , only : finidat,single_column, elm_varctl_set, iulog, noland, &
                                  scm_multcols, scm_nx, scm_ny
=======
    use elm_varctl       , only : finidat,single_column, elm_varctl_set, iulog, noland
>>>>>>> upstream/master
```

Resolution: Take HEAD. The scm_multcols, scm_nx, scm_ny symbols are used later in the lnd_init_mct subroutine.

### CODE REVIEW

File: components/eam/src/physics/cam/physpkg.F90 [hunk 1]
```diff
@@ -2387,7 +2380,7 @@ subroutine tphysbc (ztodt,               &
 
        call stratiform_tend(state, ptend, pbuf, ztodt, &
             cam_in%icefrac, cam_in%landfrac, cam_in%ocnfrac, &
-            landm, cam_in%snowhland, & ! sediment
+            cam_in%snowhland, & ! sediment
```
Comment (AaronDonahue) [inline components/eam/src/physics/cam/physpkg.F90:2383]: @singhbalwinder , I know you were dealing with an issue in scream that had to do with the land-mask.  Looking at the changes from the upstream merge, is land-mask even going to be used by the ATM model anymore.  It looks like `landm` is getting purged from all the physics processes.

**Review by AaronDonahue — APPROVED** [review]
@jgfouca , thanks for getting right on this!  Looks good to me, although I would defer to the other reviewers for how the conflicts were handled.  I'm not much of an expert on compsets.

**Review by bogensch — APPROVED** [review]
thanks once again @jgfouca .  All looks good to my eyes.

**Comment by AaronDonahue** [comment]
I think if @bogensch approves the compset changes then we should be good to go.  I readded the AUTOMERGE label.

**Comment by jgfouca** [comment]
Thanks, guys. We will need to bless CIME SCREAM tests on Monday.

**Pull request merged**